iothub_client/inc/iothub_client_properties.h

```diff
@@ -0,0 +1,298 @@
+*                                      writable properties.
+* @param[in]  payload                  Payload containing properties from Azure IoT that is to be deserialized. 
+* @param[in]  payloadLength            Length of @p payload.
+* @param[in]  componentsInModel        Optional array of components that correspond to the DTDLv2 model.  Can be NULL 
+*                                      for models that don't contain sub-components.
+* @param[in]  numComponentsInModel     Number of entries in @p componentsInModel.
+* @param[out] propertyIteratorHandle   Returned handle used for subsequent iteration calls.
+* @param[out] propertiesVersion        Returned version of the writable properties.  This is used when acknowledging a 
+*                                      write property update.
+* 
+* @remarks  Applications typically will invoke this API in their IOTHUB_CLIENT_PROPERTIES_RECEIVED_CALLBACK 
+*           implementation.  Many of the parameters this function takes should come directly from the 
+*           IOTHUB_CLIENT_PROPERTIES_RECEIVED_CALLBACK itself.
+*
+*           Applications must call @p IoTHubClient_Deserialize_Properties_DestroyIterator to free @p 
+*           propertyIteratorHandle on completion.
+*
+* @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+*/
+IOTHUB_CLIENT_RESULT IoTHubClient_Deserialize_Properties_CreateIterator(
+    IOTHUB_CLIENT_PROPERTY_PAYLOAD_TYPE payloadType,
+    const unsigned char* payload,
+    size_t payloadLength,
+    const char** componentsInModel,
+    size_t numComponentsInModel,
+    IOTHUB_CLIENT_PROPERTY_ITERATOR_HANDLE* propertyIteratorHandle,
+    int* propertiesVersion);
+
+/**
+* @brief   Gets the next property during iteration.
+*
+* @param[in]   propertyIteratorHandle   Iteration handle returned by @p IoTHubClient_Deserialize_Properties_CreateIterator.
+* @param[out]  property                 @p IOTHUB_CLIENT_DESERIALIZED_PROPERTY containing a deserialized representation 
+*                                       of the properties.
+* @param[out]  propertySpecified        Returned value indicating whether a property was found or not.  If false, this 
+*                                       indicates all components have been iterated over.
+*
+* @remarks  Applications must call @p IoTHubClient_Deserialize_Properties_DestroyProperty to free the @p property returned 
+*           by this call.
+*
+* @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+*/
+IOTHUB_CLIENT_RESULT IoTHubClient_Deserialize_Properties_GetNextProperty(
+    IOTHUB_CLIENT_PROPERTY_ITERATOR_HANDLE propertyIteratorHandle,
+    IOTHUB_CLIENT_DESERIALIZED_PROPERTY* property,
+    bool* propertySpecified);
+
+/**
+* @brief   Frees memory allocated by IoTHubClient_Deserialize_Properties.
+*
+* @param   property  IOTHUB_CLIENT_DESERIALIZED_PROPERTY initially allocated by @p IoTHubClient_Deserialize_Properties 
```

Comment:
>IoTHubClient_Deserialize_Properties [](http://example.com/codeflow?start=83&length=35)

Possibly my mistake but I am not able to find this function:
IoTHubClient_Deserialize_Properties.

Is this typo as the only possible allocation by sdk can be done by IoTHubClient_Deserialize_Properties_GetNextProperty, but that to do not take double pointer and expect user to allocate it.